luigi/task.py

```diff
@@ -817,19 +817,16 @@ def getpaths(struct):
     if isinstance(struct, Task):
         return struct.output()
     elif isinstance(struct, dict):
-        r = {}
-        for k, v in six.iteritems(struct):
-            r[k] = getpaths(v)
-        return r
+        return struct.__class__((k, getpaths(v)) for k, v in six.iteritems(struct))
```

Comment:
No, I think the current code is better. If I have to choose between supporting custom frozen dicts or `collections.Counter`. I would pick the former any day (for this instance)